src/HttpClientRequestAdapter.cs

```diff
@@ -365,32 +365,32 @@ private async Task ThrowIfFailedResponse(HttpResponseMessage response, Dictionar
 
             throw ex;
         }
-        private static IResponseHandler GetResponseHandler(RequestInformation requestInfo)
+        private static IResponseHandler? GetResponseHandler(RequestInformation requestInfo)
         {
             return requestInfo.GetRequestOption<ResponseHandlerOption>()?.ResponseHandler;
         }
-        private async Task<IParseNode> GetRootParseNode(HttpResponseMessage response)
+        private async Task<IParseNode?> GetRootParseNode(HttpResponseMessage response)
         {
             using var span = activitySource?.StartActivity(nameof(GetRootParseNode));
             var responseContentType = response.Content?.Headers?.ContentType?.MediaType?.ToLowerInvariant();
             if(string.IsNullOrEmpty(responseContentType))
                 return null;
-            using var contentStream = await response.Content.ReadAsStreamAsync();
-            var rootNode = pNodeFactory.GetRootParseNode(responseContentType, contentStream);
+            using var contentStream = await response.Content!.ReadAsStreamAsync();

```

Comment:
```suggestion
            using var contentStream = await (response.Content.ReadAsStreamAsync() ?? Task.FromResult(Stream.Null));
```

what do you think of this variation instead?

Reply:
It's much clearer/explicit to be honest. I was inferring from the above that the response Content can't be null since we have been able to extract the content type at this point successfully.